odilia/src/tower/service_ext.rs

```diff
@@ -0,0 +1,44 @@
+use crate::tower::{
+	async_try::{AsyncTryInto, AsyncTryIntoLayer, AsyncTryIntoService},
+	iter_svc::IterService,
+	state_svc::{StateLayer, StateService},
+	sync_try::{TryIntoLayer, TryIntoService},
+	unwrap_svc::UnwrapService,
+};
+use std::{convert::Infallible, future::Future, sync::Arc};
+use tower::{util::MapResult, Layer, Service, ServiceExt as TowerServiceExt};
+
+pub trait ServiceExt<Request>: Service<Request> {
+	fn request_try_from<I, R, Fut1>(self) -> TryIntoService<Request, I, Self, R, Fut1>
```

Comment:
aha, so this is where that method came from. Maybe add dock comments explaining what this does?

Reply:
Good plan. Lots of docs to go if I'm honest.

Reply:
yep, perhaps we should consider writing more docks before we're fully able to merge this, but we're getting there